src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/SupplementaryEvidenceMapperTest.java

```diff
@@ -18,45 +20,84 @@
     private static final SupplementaryEvidenceMapper mapper = new SupplementaryEvidenceMapper();
 
     @Test
-    public void from_envelope_maps_all_fields_correctly() {
+    public void maps_all_fields_correctly() {
         // given
-        Envelope envelope = SampleData.envelope(5);
+        List<Document> existingDocs =
+            asList(
+                new Document("a.pdf", "aaa", "type_a", "subtype_a", now().plusSeconds(1), "http://localhost/a.pdf"),
+                new Document("b.pdf", "bbb", "type_b", "subtype_b", now().plusSeconds(2), "http://localhost/b.pdf")
+            );
+
+        List<Document> envelopeDocs =
+            asList(
+                new Document("x.pdf", "xxx", "type_x", "subtype_x", now().plusSeconds(3), "http://localhost/x.pdf"),
+                new Document("y.pdf", "yyy", "type_y", "subtype_y", now().plusSeconds(4), "http://localhost/y.pdf")
+            );
 
         // when
-        SupplementaryEvidence result = mapper.mapEnvelope(envelope);
+        SupplementaryEvidence result = mapper.map(existingDocs, envelopeDocs);
 
         // then
         assertThat(result.evidenceHandled).isEqualTo("No");
-        assertThat(result.scannedDocuments).hasSize(5);
+        assertThat(result.scannedDocuments).hasSize(4);
         assertThat(result.scannedDocuments)
             .extracting(ccdDoc ->
                 tuple(
                     ccdDoc.value.fileName,
                     ccdDoc.value.controlNumber,
                     ccdDoc.value.type,
                     ccdDoc.value.subtype,
-                    ccdDoc.value.url.documentUrl,
-                    ccdDoc.value.scannedDate
+                    ccdDoc.value.scannedDate,
+                    ccdDoc.value.url.documentUrl
                 ))
-            .containsExactlyElementsOf(envelope.documents.stream().map(envDoc ->
-                tuple(
-                    envDoc.fileName,
-                    envDoc.controlNumber,
-                    envDoc.type,
-                    envDoc.subtype,
-                    envDoc.url,
-                    toLocalDateTime(envDoc.scannedAt)
-                )).collect(toList())
+            .containsExactly(
+                tuple("a.pdf", "aaa", "type_a", "subtype_a", toLocalDateTime(existingDocs.get(0).scannedAt), "http://localhost/a.pdf"),
+                tuple("b.pdf", "bbb", "type_b", "subtype_b", toLocalDateTime(existingDocs.get(1).scannedAt), "http://localhost/b.pdf"),
+                tuple("x.pdf", "xxx", "type_x", "subtype_x", toLocalDateTime(envelopeDocs.get(0).scannedAt), "http://localhost/x.pdf"),
+                tuple("y.pdf", "yyy", "type_y", "subtype_y", toLocalDateTime(envelopeDocs.get(1).scannedAt), "http://localhost/y.pdf")
+            );
+    }
+
+    @Test
+    public void should_not_add_document_from_envelope_if_its_already_present() {
```

Comment:
We could make this test a bit clearer about what it means that the document is already present.

Reply:
done